benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/scheduler/ExperimentTaskScheduler.java

```diff
@@ -155,10 +158,19 @@ public synchronized void handleRunningExperiment(String experimentID) {
           break;
         }
 
+        runningState = experimentModelDAO.getRunningState(experimentID);
+        logger.info("handleRunningExperiment: prevExperimentState == " + prevExperimentState);
+        logger.info("handleRunningExperiment: runningState == " + runningState);
+
         // Exit as soon as final state is executed
-        if (prevTestState != null && experimentState != null
-            && prevTestState.name().equals(experimentState.name())
-            && experimentState == TERMINATED) {
+        if (prevExperimentState == TERMINATED) {
```

Comment:
You can use `experimentState` because `TERMINATED` is a leaf state. You do it in https://github.com/benchflow/benchflow/pull/534/files#diff-e320626f292c5fd5e3bf5ea062ab6c90R285